packages/front-end/components/Settings/EditDataSourceSettingsForm.tsx

```diff
@@ -269,6 +280,20 @@ const EditDataSourceSettingsForm: FC<{
                               )
                             }
                           />
+                          <div>
+                            <label style={{ padding: 3 }}>
+                              Add Experiment & Variation Name
+                            </label>
+                            <Toggle
+                              id="nameToggle"
+                              value={addNameCols}
+                              type="environment"
+                              setValue={() => {
+                                nameColToggle();
```

Comment:
Instead of using `nameColToggle` and `addNameCols`, just use the form value directly.

Also, don't use `type="environment"`.  That's reserved for feature environment toggles only.  For normal form toggles, you can either get rid of the `type` completely or specify a value of `"toggle"`.

```tsx
<Toggle
  id="nameToggle"
  value={form.watch(`settings.queries.exposure.${i}.hasNameCol`) || false}
  setValue={(hasNameCol) => {
    form.setValue(`settings.queries.exposure.${i}.hasNameCol`, hasNameCol);
  }}
/>
```

Reply:
Also, we should update the list of `Required columns` below if this is enabled and add `experiment_name` and `variation_name` to the list.

Reply:
Is the current update process good enough? Since we're not using state, the list doesn't directly update as soon as you click the toggle and add the name columns to the query. But, after you save it, it does appear in the column list when you go back and try to edit again